### ISSUE TITLE
Add setter for "standalone" attribute in StaxEventItemWriter [BATCH-2856]

### ISSUE THREAD

**Issue by spring-issuemaster** [spring-projects/spring-batch#758]
**[Mahmoud Ben Hassine](https://jira.spring.io/secure/ViewProfile.jspa?name=mbenhassine)** opened **[BATCH-2856](https://jira.spring.io/browse/BATCH-2856?redirect=false)** and commented

As of v4.2.0, the only way to add the `standalone` attribute in the XML declaration of the output file of `StaxEventItemWriter` is to extend the writer and override `startDocument` (as stated in its [javadoc](https://docs.spring.io/spring-batch/docs/4.2.x/api/org/springframework/batch/item/xml/StaxEventItemWriter.html#startDocument-javax.xml.stream.XMLEventWriter-)). Here is an example:

```java
@Bean
public StaxEventItemWriter<Person> itemWriter() {
    StaxEventItemWriter<Person> itemWriter = new StaxEventItemWriter<Person>() {
        @Override
        protected void startDocument(XMLEventWriter writer) throws XMLStreamException {
            writer.add(createXmlEventFactory().createStartDocument(DEFAULT_ENCODING, DEFAULT_XML_VERSION, true));
            writer.add(xmlEventFactory.createStartElement(getRootTagNamespacePrefix(), getRootTagNamespace(), getRootTagName()));
        }
    };
    // TODO set other properties on the writer
    return itemWriter;
}
```

The `StaxEventItemWriter` provides setters for the `version` and `encoding` attributes, but not for the `standalone` attribute. I believe it can also provide a setter for `standalone` as this will make it easier to customize the XML declaration rather than having to extend the writer and override `startDocument` (which is error prone BTW if we forget to add the root element as shown in the example above).

NB: The example shown above works with Java 9+, see https://bugs.openjdk.java.net/browse/JDK-8139584 

---

**Affects:** 4.2.0

**Reference URL:** https://stackoverflow.com/questions/58714744/staxeventitemwriter-xml-header

**Comment by parikshitdutta** [comment]
Hi @mminella, just wanted to check with you, if you are active on this one, or I can give it a try.

Thank you.

**Comment by mminella** [comment]
Hi @parikshitdutta I assigned this to you as your requested. Let me know if you don't have time to work on it.

**Comment by parikshitdutta** [comment]
Hi @mminella,
I was thinking PR#3752 to pass through so that there is no effort in conflict resolution, Please suggest.

Thank you.

**Comment by parikshitdutta** [comment]
Thanks @mminella, I will take care of it.

**Comment by parikshitdutta** [comment]
Hi @mminella,
To complete this task, it seems, we need to upgrade ```stax2-api``` from ```4.2``` to ```4.2.1```, in our case ```woodstox-core``` from ```6.2.0``` to ```6.2.1```

Reason:
```stax2-api-4.2``` ignores ```standalone``` attribute completely, check here:
https://github.com/FasterXML/stax2-api/blob/1bcbcf1e81a9820adfe45acbee94ae65187c72f7/src/main/java/org/codehaus/stax2/ri/Stax2EventWriterImpl.java#L84-L89

whereas, ```stax2-api-4.2.1``` seems to fix this part, check here:
https://github.com/FasterXML/stax2-api/blob/8f10bb50124cb70048a3ec8dd2dca4421a04733c/src/main/java/org/codehaus/stax2/ri/Stax2EventWriterImpl.java#L84-L91

Thank you.

**Comment by parikshitdutta** [comment]
Hi @mminella,
Considering the observation mentioned above (https://github.com/spring-projects/spring-batch/issues/758#issuecomment-677382726) and patch level upgrade for ```woodstox-core```, I thought it would be okay to update the build script. 

Please have a look at PR #3764, I will look forward to your feedback.

Thank you.

**Comment by benas** [comment]
Resolved with #3764 .
